### PR TITLE
Update to Electron23 and Node18

### LINKED ISSUE #7172: Update to electron 23
There is a new version of electron and it is now included in nixpkgs so we probably should update.

* https://search.nixos.org/packages?channel=22.11&show=electron&from=0&size=50&sort=relevance&type=packages&query=electron
* https://www.electronjs.org/blog/electron-22-0

### LINKED ISSUE #7609: Dependencies: Node to v18
After updating Node to v18 we can
- https://github.com/trezor/trezor-suite/issues/5564 remove this lib and `cross-fetch`
- update Electron to v23

### PR DESCRIPTION
**Pull request by tomasklim**
## Description

- update to [Node18](https://nodejs.org/de/blog/announcements/v18-release-announce/)
- update to [Electron23](https://www.electronjs.org/blog/electron-23-0) (kills Windows 8.1 and older)
- remove `node-fetch` library
- leave `cross-fetch` in `connect` related packages as `connest` is still supported in `node14` and `node16`
- leave `cross-fetch` in packages where `fetch` is used in tests, `jest26` does not support global `fetch`, we first need to update `jest` to version `>=28` #6025
- `connect` is no longer tested in `node14`, maintanance is ending in less than 2 months anyway

## Related Issue

Closes #7172
Closes #7609

### CODE REVIEW

**Comment by Nodonisko** [comment]
Can you try to update also `react-redux`? Seems there are no real breaking changes for us anyway https://github.com/reduxjs/react-redux/releases/tag/v8.0.0

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-16.15.0
+18.14.0
```
Comment (Nodonisko) [inline .nvmrc:1]: Update is probably necessary also here https://vscode.dev/github/trezor/trezor-suite/blob/80d45b1adaf5758a138a2b79f3b7ca67351871dc/docker/ci-base/Dockerfile#L100

**Comment by tomasklim** [comment]
@SocketSecurity ignore electron@23.0.0
@SocketSecurity ignore esbuild@0.17.8

**Comment by Nodonisko** [comment]
@tomasklim What else needs to be done to get this merged? Node 18 it's recommended for latest version of React Native so it will be nice to have it updated.

**Review by Nodonisko — APPROVED** [review]

**Comment by tomasklim** [comment]
@Nodonisko @vdovhanych will update nixos environment next week  and we will be able to merge it i hope

**Comment by matejkriz** [comment]
Please do not merge it before the freeze (tomorrow)

**Review by vdovhanych — APPROVED** [review]

**Comment by tomasklim** [comment]
@SocketSecurity ignore electron@23.1.1
@SocketSecurity ignore esbuild@0.17.10

File: packages/auth-server/src/index.ts [hunk 2]
```diff
@@ -1,5 +1,4 @@
 import express from 'express';
-import fetch from 'node-fetch';
```
Comment (sime) [inline packages/auth-server/src/index.ts:2]: 🎉

**Review by Nodonisko — APPROVED** [review]

**Review by matejkriz — APPROVED** [review]
🎉 ❤️

**Comment by tomasklim** [comment]
@SocketSecurity ignore esbuild@0.17.11

**Pull request merged**